public/js/SymbolGridCalculator.js

```diff
@@ -73,35 +73,99 @@ class SymbolGridCalculator extends React.Component {
         filterPrice: { tickSize },
         quoteAsset
       },
-      symbolConfiguration
+      symbolConfiguration,
+      sell
     } = symbolInfo;
 
+    if (!sell.lastBuyPrice) {
+      return (
+        <span className='header-column-icon-wrapper grid-calculator-wrapper'>
+          <button
+            type='button'
+            className='btn btn-sm btn-link mx-1 p-0 text-white'
+            onClick={this.handleModalShow}>
+            <i className='fas fa-calculator fa-sm'></i>
+          </button>
+
+          <Modal show={this.state.showModal} onHide={this.handleModalClose}>
+            <Modal.Header className='pt-1 pb-1'>
+              <Modal.Title>Grid calculator - {symbol}</Modal.Title>
+            </Modal.Header>
+            <Modal.Body>
+              To use the calculator, you need to have an active buy grid.
+              Trigger a buy order or set the last buy price.
+            </Modal.Body>
+            <Modal.Footer>
+              <Button
+                variant='secondary'
+                size='sm'
+                onClick={this.handleModalClose}>
+                Close
+              </Button>
+            </Modal.Footer>
+          </Modal>
+        </span>
+      );
+    }
+
     const {
-      buy: { gridTrade }
+      buy: {
+        currentGridTradeIndex: buyCurrentGridTradeIndex,
+        gridTrade: buyGridTrade
+      },
+      sell: { gridTrade: sellGridTrade }
     } = symbolConfiguration;
 
+    // Detect obvious manual buys
+    // Manual buys during an active grid or outside the bot are not currently
+    // detected
+    const firstNonExecutedTradeIndex = buyGridTrade.findIndex(
```

Comment:
@rando128 
I was refactoring and writing test cases for the changes.

However, I think this change should remain in the frontend.

Firstly, the feature is not critical to the trade, but it's complex to calculate. 
Secondly, the logic code is duplicated in the frontend and backend. 

Therefore, we can put the common calculation code for the next best buy amount in the frontend rather than the backend. 

What do you think? If you are ok, I can refactor it when I get time.

Reply:
@chrisleekr, it is true there is a bit of complexity, but I don't think it is particularly intensive, computationally speaking. So there is no performance impact on the bot trading.

It is also true part of the logic is still present in the frontend because of the calculator which uses inputs from the user. We could expose an API to the front to keep the logic exclusively in the server, but I would need more time to achieve this configuration.

This real benefit of this PR will come with my next PR where I plan to add automatic buy amount calculation when a buy grid gets triggered. You can have a look at the draft code [here](https://github.com/chrisleekr/binance-trading-bot/compare/master...rando128:binance-trading-bot-pr:feat/auto-buy-amount). It's almost done but I need more time to test it live. Also I will refactor to match the way you started to refactor this PR which is quite neat.

If you feel automatic buy amount calculation is not useful for your fork, then we can move everything back in the front.

Reply:
@rando128 ohhhhh, OK. The draft code looks cool as well. Nice.

I will continue to write test cases to make sure it's covered. And I will refactor the frontend the same as what I did for the backend. 

Thanks, and sorry for slow progress. I just don't have much time. :(

Reply:
Thx @chrisleekr. No worries for the slow progress. We are all in the same boat :)